providers/gitlab_test.go

```diff
@@ -208,97 +208,102 @@ var _ = Describe("Gitlab Provider Tests", func() {
 
 	Context("when filtering on gitlab entities (groups and projects)", func() {
 		type entitiesTableInput struct {
-			expectedValue []string
-			projects      []string
-			groups        []string
+			allowedProjects []string
+			allowedGroups   []string
+			scope           string
+			expectedAuthz   bool
+			expectedError   error
+			expectedGroups  []string
+			expectedScope   string
 		}
 
 		DescribeTable("should return expected results",
 			func(in entitiesTableInput) {
 				p.AllowUnverifiedEmail = true
+				if in.scope != "" {
+					p.Scope = in.scope
+				}
 				session := &sessions.SessionState{AccessToken: "gitlab_access_token"}
 
-				err := p.AddProjects(in.projects)
-				Expect(err).To(BeNil())
-				p.SetProjectScope()
+				p.SetAllowedGroups(in.allowedGroups)
 
-				if len(in.groups) > 0 {
-					p.Groups = in.groups
+				err := p.SetAllowedProjects(in.allowedProjects)
+				if in.expectedError == nil {
+					Expect(err).To(BeNil())
+				} else {
+					Expect(err).To(MatchError(in.expectedError))
+					return
 				}
+				Expect(p.Scope).To(Equal(in.expectedScope))
 
 				err = p.EnrichSession(context.Background(), session)
+				Expect(err).To(BeNil())
+				Expect(session.Groups).To(Equal(in.expectedGroups))
 
+				authorized, err := p.Authorize(context.Background(), session)
 				Expect(err).To(BeNil())
-				Expect(session.Groups).To(Equal(in.expectedValue))
+				Expect(authorized).To(Equal(in.expectedAuthz))
 			},
 			Entry("project membership valid on group project", entitiesTableInput{
-				expectedValue: []string{"group:foo", "group:bar", "project:my_group/my_project"},
-				projects:      []string{"my_group/my_project"},
+				allowedProjects: []string{"my_group/my_project"},
+				expectedAuthz:   true,
+				expectedGroups:  []string{"foo", "bar", "project:my_group/my_project"},
```

Comment:
Groups not having the `group:` prefix would be a breaking change (minor though) -- but aligns to the architecture we are discussing for situations like this over in the `keycloak-oidc` PR: https://github.com/oauth2-proxy/oauth2-proxy/pull/1210#discussion_r637582145

Reply:
As far as I understaend this : this is implementation details not presented to end user so i'll go for harmonization :+1:

Reply:
> As far as I understand this : this is implementation details not presented to end user so i'll go for harmonization +1

See #1157.

I, for one, use `--htpasswd-user-group=group:foobar` in production and would be happy if it could not break unexpectedly. I know this is a workaround to a bug, but it would require at least an announcement nevertheless.

Reply:
Does the existing breaking change note in the CHANGELOG explain it well enough? Or do you think I should explicitly explain the `--htpasswd-user-group` implication as well (is that documented that you need to add a `group:` prefix? Or are you a power user that figured out the previous implementation quirk 😄 )